Makefile

```diff
@@ -237,3 +239,20 @@ curl -LO $${PB_REL}/download/v$${VERSION}/$${FILENAME} ; unzip $${FILENAME} -d $
 rm -rf $${TMP_DIR} ;\
 }
 endef
+
+start-smoke-tests: deploy-local-registry deploy-envoyfleet
+	@docker rm smoke-registry -f
+	@docker run -d -p 50000:5000 --name smoke-registry registry:2
+	@docker build -t localhost:50000/kusk-gateway:smoke -f ./build/manager/Dockerfile .
+	@docker push localhost:50000/kusk-gateway:smoke
+	
+deploy-local-registry: $(ENVTEST) $(KUSTOMIZE)
+	echo $(shell pwd)
+	bin/kustomize build smoketests/common  | kubectl apply -f -
+
+.PHONY: $(smoketests)
+$(smoketests): start-smoke-tests
+	$(MAKE) -C smoketests $@
+	@rm -rf smoketests/bin
+
+check-all: check-basic check-mocking
```

Comment:
check-all in the root Makefile doesn't really tell us anything. Consider renaming it to something related to smoke tests. Also make sure that the help is displayed correctly when running `make help` and that it includes this option

Reply:
We should add another section for testing. Currently `make help` returns this

```
❯ make help

Usage:
  make <target>

General
  help             Display this help.

Development
  create-env       Spin up a local development cluster with Minikube and install kusk Gateway
  deploy-envoyfleet  Deploy k8s resources for the single Envoy Fleet
  delete-env       Destroy the local development Minikube cluster
  manifests        Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
  generate         Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
  fmt              Run go fmt against code.
  vet              Run go vet against code.
  test             Run tests.
  testing          Run the integration tests from development/testing and then delete testing artifacts if succesfull.
  goproxy          Starts local goproxy docker instance for the faster builds. Make sure to set your shell environment variable (e.g. put into .bashrc "export GOPROXY=http://<your_ip_address>:8085".
  docker-images-cache  Saves locally frequently used container images and uploads them to Minikube to speed up the development.

Build
  build            Build manager binary.
  run              Run a controller from your host, proxying it inside the cluster.
  docker-build-manager  Build docker image with the manager.
  docker-build     Build docker images for all apps
  docker-build-manager-debug  Build docker image with the manager and debugger.

Deployment
  install-local    Install CRDs and Envoy into the K8s cluster specified in ~/.kube/config.
  uninstall-local  Uninstall CRDs and Envoy from the K8s cluster specified in ~/.kube/config.
  install          Install CRDs into the K8s cluster specified in ~/.kube/config.
  uninstall        Uninstall CRDs from the K8s cluster specified in ~/.kube/config. Call with ignore-not-found=true to ignore resource not found errors during deletion.
  deploy           Deploy controller to the K8s cluster specified in ~/.kube/config.
  deploy-debug     Deploy controller with debugger to the K8s cluster specified in ~/.kube/config.
  undeploy         Undeploy controller from the K8s cluster specified in ~/.kube/config.
  update           Runs deploy, docker build and restarts kusk-gateway-manager deployment to pick up the change
  update-debug     Runs Debug configuration deploy, docker build and restarts kusk-gateway-manager deployment to pick up the change
  cycle            Triggers kusk-gateway-manager deployment rollout restart to pick up the new container image with the same tag
  cycle-envoy      Triggers all Envoy pods in the cluster to restart
  envtest          Download envtest-setup locally if necessary.
  tools            Install all tools
```

Reply:
It won't work as I'm pulling names of smoketest from smoketests/Makefile.variables. 
Probably I don't know how to do it.

Smoketests are for our internal usage anyways.